### PR TITLE
fail to pop up When Use Flutter_boost

### PR DESCRIPTION
**Pull request by nicorobine**
With Flutter_boost, when registering the plug-in, [UIApplication sharedApplication].delegate. delegate.windows may also be  modified by function："- (BOOL)application:(UIApplication *)application DidFinishLaunchingWithOptions: (NSDictionary *) launchOptions"， The obtained rootViewController may not be the actual ViewController, when register this plug-in，which will cause the preview screen to fail to pop up

### CODE REVIEW

**Comment by JulianBissekkou** [comment]
This PR also fixes some issues when you have native navigation in your flutter app.
In our case we push a special splash screen before launching our flutter view controller.

We had to add a small workaround to make sure that he plugin does not hold a reference to our old SplashViewController.

**Comment by JulianBissekkou** [comment]
@crazecoder

File: android/src/main/java/com/crazecoder/openfile/OpenFilePlugin.java [hunk 1]
```diff
@@ -151,9 +151,19 @@ private boolean pathRequiresPermission() {
         }
 
         try {
-            String appDirCanonicalPath = new File(context.getApplicationInfo().dataDir).getCanonicalPath();
```
Comment (ipcjs) [inline android/src/main/java/com/crazecoder/openfile/OpenFilePlugin.java:154]: Why did you delete this line?
The app does not need permission to access its own data directory.
